### PR TITLE
Establishing the Development Workflow

### PR DESCRIPTION
**Pull request by jelster**
This PR adds basic repository content and configuration for the application.

- .gitignore
- package.json and package.lock.json with BabylonJS, Webpack dev dependencies
-`npm run scripts` for `start`, `build`, and `lint`
- webpack.common.js and plugin setup 

Opening PR now to allow early feedback

### CODE REVIEW

**Comment by jelster** [comment]
In case anyone is wondering, the similarities between this and [Raanan Weber's](https://github.com/RaananW/babylonjs-webpack-es6) BJS repository are intentional; the goal is to establish the foundation for building an ES6 web application using BJS, whereas Raanan's repository is oriented towards TypeScript

File: package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ],
+  "author": "Josh Elster",
+  "license": "BSD-3-Clause",
+  "bugs": {
+    "url": "https://github.com/jelster/space-truckers/issues"
+  },
+  "homepage": "https://github.com/jelster/space-truckers#readme",
+  "dependencies": {},
+  "devDependencies": {
+    "@babylonjs/core": "^4.2.0",
```
Comment (RaananW) [inline package.json:28]: in this kind of project it is pure semantics, but is there a reason not to put the fixed dependencies that are not related to the dev environment in `dependencies`?

File: src/startscene.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    Texture,
+    PointLight,
+    Scene,
+    StandardMaterial,
+    TrailMesh,
+    Scalar,
+    GlowLayer,
+    Animation,
+    ArcRotateCamera
+} from "@babylonjs/core";
```
Comment (RaananW) [inline src/startscene.js:14]: to enable tree shaking it is recommended to load directly from the directories instead of loading everything from @babylonjs/core. I know it sometimes feel like a hassle, but the package size will be thankful :-)

File: webpack.prod.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const { merge } = require('webpack-merge');
+const common = require('./webpack.common.js');
+
+const prodConfig = {
+    mode: "production",
+    devtool: "source-map"
```
Comment (RaananW) [inline webpack.prod.js:6]: only if you want sourcemaps in production

File: src/startscene.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
+import {
```
Comment (RaananW) [inline src/startscene.js:1]: Might make sense at a later stage to separate those functions in different files with different responsibilities. But for this state it is perfectly fine.

**Review by RaananW — APPROVED** [review]
Better late than never ;-)

Just have a few comments, but it looks great.

File: package.json (hunk 1 above)
Comment (jelster) [reply package.json:28]: I was wondering about that too, what sort of difference it makes. I guess I figured that since it's a build-time dep it belonged in devDeps. That emphasizes that it's not a runtime dependency, but you do make a point about it making it more clear what the project is about

File: src/startscene.js (hunk 2 above)
Comment (jelster) [reply src/startscene.js:14]: oy, good to know; I'll have to revisit the tree shaking docs on some of this but I think I gotcha. It did make me wonder though -- since we're already using webpack, wouldn't a relatively simple (custom, obv) loader be able to take the above and transform it to the path syntax?

File: webpack.prod.js (hunk 3 above)
Comment (jelster) [reply webpack.prod.js:6]: d'oh! That isn't what I intended - good catch.

File: src/startscene.js (hunk 4 above)
Comment (jelster) [reply src/startscene.js:1]: I've evolved this slightly in some early commits in the ch4 branch, was thinking maybe of bringing them over here into this one. Might put something in here to see what it would look like

File: package.json (hunk 1 above)
Comment (jelster) [reply package.json:28]: oh I was thinking of going with this for the versoining: ` @babylonjs/core: "^5.0.0-alpha.9 <= 5.0.0",`
